Remove THCAllocator

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65948
* #65944
* #65943
* __->__ #65942

This one is a bit weird. The class is called `THCIpcDeleter` but it
actually has nothing IPC-specific. It just converts
`std::shared_ptr` + `void*` into a `c10::DataPtr`. Instead, moving
the `DataPtr` conversion into the actual IPC code allows 2 memory
allocations to be elided by merging 3 separate deletion contexts
into one.

Differential Revision: [D31386278](https://our.internmc.facebook.com/intern/diff/D31386278)

peterbell10 committed:
Remove THCAllocator

This one is a bit weird. The class is called `THCIpcDeleter` but it
actually has nothing IPC-specific. It's just converts
`std::shared_ptr` + `void*` into a `c10::DataPtr`. Instead, moving
the `DataPtr` convertion into the actual IPC code allows 2 memory
allocations to be ellided by merging 3 separate deletion contexts
into one.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THCAllocator"


This one is a bit weird. The class is called `THCIpcDeleter` but it
actually has nothing IPC-specific. It's just converts
`std::shared_ptr` + `void*` into a `c10::DataPtr`. Instead, moving
the `DataPtr` convertion into the actual IPC code allows 2 memory
allocations to be ellided by merging 3 separate deletion contexts
into one.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THCAllocator"


This one is a bit weird. The class is called `THCIpcDeleter` but it
actually has nothing IPC-specific. It's just converts
`std::shared_ptr` + `void*` into a `c10::DataPtr`. Instead, moving
the `DataPtr` convertion into the actual IPC code allows 2 memory
allocations to be ellided by merging 3 separate deletion contexts
into one.

[ghstack-poisoned]

ngimel commented:
This one also can't be imported 
```
RuntimeError: We tried to import new changes for D31236630, but this diff has already been landed.  This could occur if you rebased your stack prior to the landed diff being pushed by ShipIt; in this case, wait until this diff is exported with ShipIt, rebase on master, and try again.
```

peterbell10 committed:
Update on "Remove THCAllocator"


This one is a bit weird. The class is called `THCIpcDeleter` but it
actually has nothing IPC-specific. It's just converts
`std::shared_ptr` + `void*` into a `c10::DataPtr`. Instead, moving
the `DataPtr` conversion into the actual IPC code allows 2 memory
allocations to be elided by merging 3 separate deletion contexts
into one.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31386278).
